oxidation/crates/parsec_serialization_format/src/lib.rs

```diff
@@ -8,13 +8,16 @@ use std::io::{BufRead, BufReader};
 use std::path::PathBuf;
 use syn::{parse_macro_input, LitStr};
 
-fn content(path: String) -> String {
+fn path_from_str(path: &str) -> PathBuf {
     let manifest_dir_path = std::env::var("CARGO_MANIFEST_DIR")
         .expect("CARGO_MANIFEST_DIR should be set")
         .parse::<PathBuf>()
         .expect("CARGO_MANIFEST_DIR must be a valid path");
-    let file_path = manifest_dir_path.join(&path);
-    let file = File::open(file_path).unwrap_or_else(|e| panic!("{e}"));
+    manifest_dir_path.join(path)
+}
+
+fn content_from_file(path: &PathBuf) -> String {
+    let file = File::open(path).unwrap_or_else(|e| panic!("{e}"));
```

Comment:
```suggestion
    let file = File::open(path).unwrap();
```
`unwrap` do exactly the same or you may want to use `expect` to add a message